src/test/TxTests.cpp

```diff
@@ -191,25 +197,47 @@ applyCheck(TransactionFramePtr tx, Application& app, bool checkSeqNum)
         }
         REQUIRE((!res || tx->getResultCode() == txSUCCESS));
 
-        // checks that the failure is the same if pre checks failed
         if (!check)
         {
-            if (tx->getResultCode() != txFAILED)
-            {
-                REQUIRE(checkResult == tx->getResult());
-            }
-            else
+            if (checkResult.result.code() == txFAILED &&
+                tx->getResultCode() == txFAILED)
             {
+                // should be equal up to first failure
                 auto const& txResults = tx->getResult().result.results();
                 auto const& checkResults = checkResult.result.results();
                 for (auto i = 0u; i < txResults.size(); i++)
                 {
-                    REQUIRE(checkResults[i] == txResults[i]);
-                    if (checkResults[i].code() == opBAD_AUTH)
+                    try
+                    {
+                        throwIf(checkResults[i]);
+                        throwIf(txResults[i]);
+                    }
+                    catch (...)
                     {
-                        // results may not match after first opBAD_AUTH
                         break;
                     }
+
+                    REQUIRE(checkResults[i] == txResults[i]);
+                }
+            }
+            else
+            {
+                auto header = ltxTx.loadHeader();
+                if (tx->getResult().result.code() != txBAD_AUTH_EXTRA &&
```

Comment:
The way this condition is written is not super clear:
rewrite it in such a way that it's easy to understand why you're blacklisting some result permutation.